pkg/scheduler/framework/plugins/nodevolumelimits/csi_test.go

```diff
@@ -598,8 +610,8 @@ func getFakeCSIStorageClassLister(scName, provisionerName string) fakeframework.
 	}
 }
 
-func getFakeCSINodeLister(csiNode *storagev1.CSINode) fakeframework.CSINodeLister {
-	if csiNode != nil {
+func getFakeCSINodeLister(csiNode *storagev1.CSINode, isNilCsiNode bool) fakeframework.CSINodeLister {
```

Comment:
It looks like the function already handles a nil `csiNode` input. Do you need to add this extra field?

Reply:
Because we cannot set the csiNode to nil before this function, there is a set up step for csiNode.spec.Something. The setup will fail?

Reply:
I just tried to modify the test case for limitSource as node in the **master branch**:

```
{
			newPod:           inTreeOneVolPod,
			existingPods:     []*v1.Pod{inTreeTwoVolPod},
			filterName:       "csi",
			maxVols:          2,
			driverNames:      []string{csilibplugins.AWSEBSInTreePluginName, ebsCSIDriverName},
			migrationEnabled: true,
			limitSource:      "node",
			test:             "should count in-tree volumes if migration is enabled",
		},
```
The test just passed

Reply:
The test case should use a pod specification using an inline volume.

Also it looks like the FakeLister doesn't return nil in a "not found" scenario: https://github.com/kubernetes/kubernetes/blob/6ec579904c6120b961ed0a1752cd9f92dfb5124c/pkg/scheduler/framework/fake/listers.go#L264

Reply:
inTreeInlineVolPod on master branch testing also did not trigger the painc. Made the change for the fake lister, some other existing tests start to fail:

```
    --- FAIL: TestCSILimits/should_count_in-tree_volumes_if_migration_is_enabled (0.00s)
    --- FAIL: TestCSILimits/should_count_unbound_in-tree_volumes_if_migration_is_enabled (0.00s)
    --- FAIL: TestCSILimits/should_count_in-tree_inline_volumes_if_migration_is_enabled (0.00s)
    --- FAIL: TestCSILimits/should_count_in-tree_and_csi_volumes_if_migration_is_enabled_(when_scheduling_in-tree_volumes) (0.00s)
    --- FAIL: TestCSILimits/should_count_in-tree,_inline_and_csi_volumes_if_migration_is_enabled_(when_scheduling_in-tree_volumes) (0.00s)
    --- FAIL: TestCSILimits/should_count_in-tree_and_csi_volumes_if_migration_is_enabled_(when_scheduling_csi_volumes) (0.00s)

```